src/rfm.rs

```diff
@@ -353,6 +376,46 @@ where
         self.write(Registers::TestLna, boost as u8)
     }
 
+    pub fn tx_level(&mut self, power_level: i8) -> Result<(), Espi> {
+        let adjusted_power = power_level.clamp(-18, 13);
+
+        let reg_value = PaOptions::Pa0On as u8 | (adjusted_power + 18) as u8;
+
+        self.write(Registers::PaLevel, reg_value)?;
+        self.tx_pwr = adjusted_power;
+        Ok(())
+    }
+
+    pub fn tx_level_high_pwr(&mut self, power_level: i8) -> Result<(), Espi> {
```

Comment:
Same as above with the function documentation and the name. Also throughout the whole function, please convert all "magic" numbers into constants.